contracts/mainnet/connectors/uniswapV3/helpers.sol

```diff
@@ -0,0 +1,246 @@
+            memory params = INonfungiblePositionManager.MintParams(
+                address(_token0),
+                address(_token1),
+                params.fee,
+                params.tickLower,
+                params.tickUpper,
+                _amount0,
+                _amount1,
+                _minAmt0,
+                _minAmt1,
+                address(this),
+                block.timestamp
+            );
+        (tokenId, liquidity, amount0, amount1) = nftManager.mint(params);
+    }
+
+    /**
+     * @dev Check if token address is etherAddr and convert it to weth
+     */
+    function _checkETH(
+        uint256 _tokenId,
+        uint256 _amount0,
+        uint256 _amount1
+    ) internal {
+        (, , address token0, address token1, , , , , , , , ) = nftManager
+            .positions(_tokenId);
+        uint256 isEth = token0 == wethAddr ? 0 : 2;
+        isEth = token0 == wethAddr ? 1 : 2;
+        convertEthToWeth(isEth == 0, TokenInterface(token0), _amount0);
+        convertEthToWeth(isEth == 1, TokenInterface(token1), _amount1);
+        approve(TokenInterface(token0), address(nftManager), _amount0);
+        approve(TokenInterface(token1), address(nftManager), _amount1);
+    }
+
+    /**
+     * @dev addLiquidity function which interact with Uniswap v3
+     */
+    function _addLiquidity(
+        uint256 _tokenId,
+        uint256 _amount0,
+        uint256 _amount1,
+        uint256 _amount0Min,
+        uint256 _amount1Min
+    )
+        internal
+        returns (
+            uint128 liquidity,
+            uint256 amount0,
+            uint256 amount1
+        )
+    {
+        _checkETH(_tokenId, _amount0, _amount1);
+        INonfungiblePositionManager.IncreaseLiquidityParams
+            memory params = INonfungiblePositionManager.IncreaseLiquidityParams(
+                _tokenId,
+                _amount0,
+                _amount1,
+                _amount0Min,
+                _amount1Min,
+                block.timestamp
+            );
+        (liquidity, amount0, amount1) = nftManager.increaseLiquidity(params);
+    }
+
+    /**
+     * @dev decreaseLiquidity function which interact with Uniswap v3
+     */
+    function _decreaseLiquidity(
+        uint256 _tokenId,
+        uint128 _liquidity,
+        uint256 _amount0Min,
+        uint256 _amount1Min
+    ) internal returns (uint256 amount0, uint256 amount1) {
+        INonfungiblePositionManager.DecreaseLiquidityParams
+            memory params = INonfungiblePositionManager.DecreaseLiquidityParams(
+                _tokenId,
+                _liquidity,
+                _amount0Min,
+                _amount1Min,
+                block.timestamp
+            );
+        (amount0, amount1) = nftManager.decreaseLiquidity(params);
+    }
+
+    /**
+     * @dev collect function which interact with Uniswap v3
+     */
+    function _collect(
+        uint256 _tokenId,
+        uint128 _amount0Max,
+        uint128 _amount1Max
```

Comment:
@cryptoDev222 Let's keep these values as `uint256` only and convert them internal to `uint128`

Reply:
Okay, got it.